### DISCUSSION TITLE
Material discussion

### DISCUSSION THREAD

**Discussion by bakerpdgit** [bakerpdgit/tutorial3#1] [Announcements]
Share info here about which material you are currently working on so that we don't duplicate our efforts.

**Comment by Anonymous12358** [comment]
Now that my exams are finished, I though I could contribute some more functional programming content? It may be more difficult to motivate a functional programming style in a language with imperative constructs, but not impossible. It would also go beyond A-level, but this repository is separate from the main A-level content anyway.

**Reply by Anonymous12358** [reply]
Is there any chance of Python Sponge updating to version 3.13?
I want to make a `let` statement to handle variables in a purely functional way:
```python3
with let(x=42):
  print(x)
```
but this requires modifying local variables in a way that, as far as I can tell, is only consistently possible in Python 3.13.
Obviously this black magic won't be exposed to students but if it's too much or you don't want to update I'll try and find another way to go.
